### PR TITLE
 Optimize getTransactionByHash for dev mode

### LINKED ISSUE #968: Optimize `getTransactionByHash` for dev mode
In order to make execution of tests with local-node (for example) faster, we still need to save some time from `getTransactionByHash`, because hardhat (and ethers) is using this endpoint to ping for tx hash. We can potentially save around 200ms, if we can optimize this method, only for dev mode.

### PR DESCRIPTION
**Pull request by natanasow**
**Description**:

**Related issue(s)**:

Fixes #968

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/lib/clients/mirrorNodeClient.ts [hunk 1]
```diff
@@ -121,12 +122,12 @@ export class MirrorNodeClient {
...
         //@ts-ignore
         axiosRetry(axiosClient, {
-            retries: parseInt(process.env.MIRROR_NODE_RETRIES!) || 3,
+            retries: isDevMode ? 5 : parseInt(process.env.MIRROR_NODE_RETRIES!) || 3,
             retryDelay: (retryCount, error) => {
                 const request = error?.request?._header;
                 const requestId = request ? request.split('\n')[3].substring(11,47) : '';
                 const requestIdPrefix = formatRequestIdMessage(requestId);
-                const delay = (parseInt(process.env.MIRROR_NODE_RETRY_DELAY!) || 250) * retryCount;
+                const delay = isDevMode ? 200 : (parseInt(process.env.MIRROR_NODE_RETRY_DELAY!) || 250) * retryCount;
```
Comment (Nana-EC) [inline packages/relay/src/lib/clients/mirrorNodeClient.ts:130]: Add a `MIRROR_NODE_RETRY_DELAY_DEVMODE`
```suggestion
                const delay = isDevMode ? MIRROR_NODE_RETRY_DELAY_DEVMODE : (parseInt(process.env.MIRROR_NODE_RETRY_DELAY!) || 250) * retryCount;
```

Comment (Nana-EC) [inline packages/relay/src/lib/clients/mirrorNodeClient.ts:130]: Add a `MIRROR_NODE_RETRIES_DEVMODE`
```suggestion
                const delay = isDevMode ? MIRROR_NODE_RETRIES_DEVMODE : (parseInt(process.env.MIRROR_NODE_RETRY_DELAY!) || 250) * retryCount;
```

**Review by Nana-EC — COMMENTED** [review]
LG. 
Might as well make it configurable for devs.
Add in the appropriate places. Can skip production paths

**Review by georgi-l95 — APPROVED** [review]
LG. This improves E2E execution with around 10%.

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**